packages/core/src/document.ts

```diff
@@ -584,56 +585,55 @@ class Document extends EventEmitter implements DocStateHolder {
   /**
    * Request anchor for the latest document state
    */
-  async anchor (): Promise<void> {
+  anchor(): void {
     // eslint-disable-next-line @typescript-eslint/no-this-alias
     const doc = this
     const requestTip: CID = this.tip
 
-    this._context.anchorService.on(this.id.toString(), async function listener (asr: AnchorServiceResponse) {
-      if (!asr.cid.equals(requestTip)) {
-        // This message is about a different tip for the same document
-        return
-      }
-      switch (asr.status) {
-        case 'PENDING': {
-          const state = doc._doctype.state
-          state.anchorScheduledFor = asr.anchorScheduledFor
-          doc._doctype.state = state
-          await doc._updateStateIfPinned()
-          return
-        }
-        case 'PROCESSING': {
-          const state = doc._doctype.state
-          state.anchorStatus = AnchorStatus.PROCESSING
-          doc._doctype.state = state
-          await doc._updateStateIfPinned()
-          return
-        }
-        case 'COMPLETED': {
-          doc._context.anchorService.removeListener(doc.id.toString(), listener)
-
-          await doc._handleTip(asr.anchorRecord)
-          await doc._updateStateIfPinned()
-          await doc._publishTip()
-          return
-        }
-        case 'FAILED': {
-          doc._context.anchorService.removeListener(doc.id.toString(), listener)
-          if (requestTip !== doc.tip) {
-            return
-          }
-
-          const state = doc._doctype.state
-          state.anchorStatus = AnchorStatus.FAILED
-          doc._doctype.state = state
-          return
-        }
-      }
-    })
-    await this._context.anchorService.requestAnchor(this.id.toString(), this.tip)
-    const state = this._doctype.state
-    state.anchorStatus = AnchorStatus.PENDING
-    this._doctype.state = state
+    const anchorStatus$ = this._context.anchorService.requestAnchor(this.id.baseID, this.tip);
+    const subscription = anchorStatus$
+        .pipe(
+            concatMap(async (asr) => {
```

Comment:
Unlike `mergeMap`, concatMap stops the feed propagation further until the function inside `concatMap` is done.
`interval` inside EthereumAnchorService pushes events every N seconds. I would like to be sure, that processing of those happens sequentially, instead of having multiple request-processing instances at a time.

Reply:
interesting.  So `pipe` doesn't process the events sequentially already?  It just seems weird to me since the docs for `concatMap` all talk about merging multiple input observables into one output observable, but here we only have 1 input observable.  Is there no more specific and semantically meaningful operator to take a single observable and just tell it to finish processing one event before starting to process the next?